packages/headless/src/controllers/smart-snippet-questions-list/headless-smart-snippet-questions-list.ts

```diff
@@ -120,13 +173,66 @@ export function buildSmartSnippetQuestionsList(
     );
   };
 
+  let lastSearchResponseId: string | null = null;
```

Comment:
This is getting pretty hard to read...

Might benefit from some good old "extract private function with descriptive names" ?